### DISCUSSION TITLE
Provider preference for openrouter

### DISCUSSION THREAD

**Discussion by Bigsy** [cline/cline#1076] [Feature Requests]
As some providers offer better context length eg together.ai 128k for deepseek (64k otherwise) - Hyperbolic 128k for qwen coder etc and openrouter offer a way to choose provider it would be good to expose that so we can pick the best one for Cline eg

```
{
  "model": "deepseek/deepseek-chat",
  "messages": [
    {"role": "user", "content": "Hello"}
  ],
  "provider": {
    "order": ["Together"],
    "allow_fallbacks": false
  }
}
```

**Comment by kedihacker** [comment]
I want the full experience UI could just be raw options
